sdk/include/opentelemetry/sdk/metrics/view/attributes_processor.h

```diff
@@ -0,0 +1,34 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/sdk/common/attribute_utils.h"
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+using MetricAttributes = opentelemetry::sdk::common::AttributeMap;
+
+class AttributesProcessor
+{
+public:
+  virtual MetricAttributes process(
+      const opentelemetry::common::KeyValueIterable &attributes) noexcept = 0;
+};
+
+class DefaultAttributesProcessor : public AttributesProcessor
+{
+  MetricAttributes process(
```

Comment:
IIUC - by necessity, this API will need to take in an Iterator and return a concrete object.  It's likely this would get used on the "hot path" of metrics.  Does it make sense to have this API work only against `AttributeMap`, that way in the event you're just passing it through, there isn't a sort + unique step (I forget the details of attribute map).

Reply:
Yes, your understanding is correct here. The metrics API takes non-owning KeyValueIterable object as [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/9dd3da33b03fac66af275c417938e0dc69f4eda6/api/include/opentelemetry/metrics/sync_instruments.h#L40), and the plan is to make a copy of this only when required in hot-path. Attribute-processor could be one of the places to create a copy in case of passthrough.

Reply:
I expect this to (commonly) be a passthrough case.   Can deal with it after initial implementation :)